### PR TITLE
DP-1554 social media links / icons - a11y

### PR DESCRIPTION
**Pull request by jesconstantine**
This PR adds visually hidden link text to social media links so that assistive technology can provide context for the link target. (Previously, the only link content was the icon) . It also tells assistive tech to ignore the social media icons.

Ticket: https://jira.state.ma.us/browse/DP-1554

Relates to massgov/mass PR:  https://github.com/massgov/mass/pull/403

## To Test

1. Pull branch down
2. move into styleguide `cd styleguide`
3. build mayflower `gulp`
4. browse to any page pattern
5. open dev tools, inspect the link and verify that you can see link text semantically (and that markup matches the screenshot below), close dev tools
7. Scroll down and select the text of one of the footer links (to bring focus down there)
6. Turn on screen reader (make sure your volume is not muted)
  - macos: https://www.apple.com/voiceover/info/guide/_1131.html
  - windows: (if you don't have a jaws license), try chrome vox: https://chrome.google.com/webstore/detail/chromevox/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en
8. Press TAB to move through the footer links, the the footer logo, and finally to the social media links menu
9. Confirm that the link text is announced: Follow us on ________

### Expanded social media menu link and svg markup
dp-1554-social-media-menu-links

### CODE REVIEW

**Comment by jesconstantine** [comment]
@legostud I know you mentioned a gulp task for creating icon twig templates and I know this affects the icon-links pattern -- can we talk about those as well?

**Comment by gender404** [comment]
if you run gulp svg2twig in the command prompt within styleguide, it would convert all the svg files in svg-icons into svg-[name].twig files within the atoms/05-icons folder.

File: styleguide/source/_patterns/01-atoms/05-icons/svg-blog.twig [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+<svg role="presentation" id="SvgjsSvg1017" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs" width="21" height="20" viewBox="0 0 21 20"><defs id="SvgjsDefs1018"></defs><path id="SvgjsPath1019" d="M289.47 70.2937C289.47 68.7988 288.24800000000005 67.5875 286.74 67.5875C285.231 67.5875 284.008 68.7988 284.008 70.2937C284.008 71.7887 285.231 73 286.74 73C288.248 73 289.47 71.7887 289.47 70.2937ZM297.315 73C297.262 65.7413 291.334 59.8688 284.008 59.8175V63.8262C289.091 63.8775 293.216 67.9638 293.267 73ZM300.123 73C300.097 64.1713 292.895 57.0488 284.008 57.01V53C295.135 53.0388 304.145 61.9737 304.16999999999996 73Z " fill-opacity="1" transform="matrix(1,0,0,1,-284,-53)"></path></svg>
```
Comment (gender404) [inline styleguide/source/_patterns/01-atoms/05-icons/svg-blog.twig:1]: These changes should be made in the svg file not in the twig file.  run gulp svg2twig to generate the twig files.

Comment (jesconstantine) [reply styleguide/source/_patterns/01-atoms/05-icons/svg-blog.twig:1]: Thanks, done in https://github.com/massgov/mayflower/pull/320/commits/6ec718866db1e1ca40037784d396ad349066d7fd -- I looked through the gulp tasks earlier but my branch was out of date at the time so I couldn't find it! 👌

File: styleguide/source/_patterns/02-molecules/social-links.twig [hunk 2]
```diff
@@ -4,6 +4,7 @@
       {% for item in socialLinks.items %}
         <li class="ma__social-links__item">
           <a href="{{ item.href }}" class="ma__social-links__link">
+            <span class="visually-hidden">{{ item.text }}</span>
```
Comment (jesconstantine) [inline styleguide/source/_patterns/02-molecules/social-links.twig]: @legostud Does this constitute a breaking change?  Should I make it optional?  I'm working on the Drupal implementation as well: https://github.com/massgov/mass/pull/408

Comment (gender404) [reply styleguide/source/_patterns/02-molecules/social-links.twig]: I would wrap it in an if statement with a note "backward compatible for v3.1".  In the docs, I'd leave it as required

Comment (jesconstantine) [reply styleguide/source/_patterns/02-molecules/social-links.twig]: 👍 Sounds good.  Thanks - addressed in https://github.com/massgov/mayflower/pull/320/commits/f295ae0a5a9b577370e80d8001c88c318896c454

**Review by gender404 — APPROVED** [review]

File: styleguide/source/_patterns/02-molecules/social-links.md [hunk 3]
```diff
@@ -18,6 +19,8 @@ Description: displays a list of sitewide social links
         type: string(url)/required,
       icon: 
         type: string(path to icon)/required
+      altText:
+        type: string/required
```
Comment (GBirch) [inline styleguide/source/_patterns/02-molecules/social-links.md:23]: I'm cool with marking this as required, but note that we are not, in fact, treating it that way (for backwards compatibility, I assume).

Comment (jesconstantine) [reply styleguide/source/_patterns/02-molecules/social-links.md:23]: Thanks @GBirch -- @legostud came to the same conclusion.

**Pull request merged**